src/components/NostoProvider.tsx

```diff
@@ -86,7 +86,7 @@ export default function NostoProvider(props: NostoProviderProps) {
       window.nostojs(api => api.setAutoLoad(false))
     }
 
-    if (!document.querySelectorAll("[nosto-client-script]").length && !shopifyMarkets) {
+    if (!window.nosto && !shopifyMarkets) {
```

Comment:
Is this window.nosto or window.nostojs? 🤔

Reply:
window.nosto - nosto iframe window namespace
window.nostojs - loader function to register custom API callbacks

Reply:
window.nosto usage is intentional here, but I can extract it into a function call to make the intention of this check clearer

Reply:
Yes, it's confusing right now. Because, the first check is about `window.nostojs` and then we load the Nosto.js script. If the script loading fails, then window.nostojs initialization could cause issues.

Reply:
updated